server/plugin/api_test.go

```diff
@@ -977,58 +963,140 @@ func TestGetUserChannelsForTeam(t *testing.T) {
+		})
+	}
+}
+
 func TestHandleGetGitRepositoryBranches(t *testing.T) {
 	defer monkey.UnpatchAll()
-	p := Plugin{}
 	mockAPI := &plugintest.API{}
 	mockCtrl := gomock.NewController(t)
 	mockedClient := mocks.NewMockClient(mockCtrl)
-	p.API = mockAPI
-	p.Client = mockedClient
+	p := setupMockPlugin(mockAPI, nil, mockedClient)
 	for _, testCase := range []struct {
-		description                 string
-		Organization                string
-		Repository                  string
-		Project                     string
-		GetGitRepositoryBranchesErr error
-		statusCode                  int
+		description                      string
+		organization                     string
+		repository                       string
+		project                          string
+		getGitRepositoryBranchesErr      error
+		statusCode                       int
+		getGitRepositoryBranchesResponse *serializers.GitBranchesResponse
+		expectedResponse                 []*serializers.GitBranch
+		expectedErrorResponse            interface{}
 	}{
 		{
 			description:  "HandleGetGitRepositoryBranches: valid",
-			Organization: "mockOrganization",
-			Project:      "mockProject",
-			Repository:   "mockRepository",
+			organization: "mockOrganization",
+			project:      "mockProject",
+			repository:   "mockRepository",
 			statusCode:   http.StatusOK,
+			getGitRepositoryBranchesResponse: &serializers.GitBranchesResponse{
+				Value: []*serializers.GitBranch{
+					{
+						ID:   "mockID-1",
+						Name: "refs/heads/mockName-1",
+					},
+					{
+						ID:   "mockID-2",
+						Name: "refs/heads/mockName-2",
+					},
+				},
+			},
+			expectedResponse: []*serializers.GitBranch{
+				{
+					ID:   "mockID-1",
+					Name: "mockName-1",
+				},
+				{
+					ID:   "mockID-2",
+					Name: "mockName-2",
+				},
+			},
 		},
 		{
-			description:  "HandleGetGitRepositoryBranches: Invalid organization, project or repository params",
-			Organization: "mockOrganization",
-			Project:      "mockProject",
-			statusCode:   http.StatusBadRequest,
+			description:           "HandleGetGitRepositoryBranches: Invalid organization, project or repository params",
+			organization:          "mockOrganization",
+			project:               "mockProject",
+			statusCode:            http.StatusBadRequest,
+			expectedErrorResponse: map[string]interface{}{"Error": "Invalid organization, project or repository params"},
 		},
 		{
 			description:                 "HandleGetGitRepositoryBranches: GetGitRepositoryBranches returns error",
-			Organization:                "mockOrganization",
-			Project:                     "mockProject",
-			Repository:                  "mockRepository",
-			GetGitRepositoryBranchesErr: errors.New("failed to get git repository branches"),
+			organization:                "mockOrganization",
+			project:                     "mockProject",
+			repository:                  "mockRepository",
+			getGitRepositoryBranchesErr: errors.New("failed to get git repository branches"),
```

Comment:
Use constant here also.

Reply:
no constant created for this

Reply:
I can create a constant in the testcase itself